testsupport/wait/host.go

```diff
@@ -2166,3 +2166,40 @@ func EncodeUserIdentifier(subject string) string {
 
 	return encoded
 }
+
+// CreateSpaceAndSpaceBinding creates a space and spacebindig and waits until both are present.
+// We are creating both of them (Space and SpaceBinding) at the same time , with polling logic, so that we mitigate the issue with spacecleanup_controller deleting the Space before we create it's SpaceBinding.
+func (a *HostAwaitility) CreateSpaceAndSpaceBinding(t *testing.T, mur *toolchainv1alpha1.MasterUserRecord, space *toolchainv1alpha1.Space, spaceRole string, terminatingSpace toolchainv1alpha1.Condition) (*toolchainv1alpha1.Space, *toolchainv1alpha1.SpaceBinding, error) {
+	var spaceBinding *toolchainv1alpha1.SpaceBinding
+	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
+		// create the space
+		if err := a.CreateWithCleanup(t, space); err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		// create spacebinding request immediately after ...
+		spaceBinding = spacebinding.NewSpaceBinding(mur, space, spaceRole)
+		if err := a.CreateWithCleanup(t, spaceBinding); err != nil {
+			return false, err
+		}
+		// let's see if space was provisioned as expected
+		space, err = a.WaitForSpace(t, space.Name)
+		if err != nil {
+			return false, err
+		}
+		if test.ContainsCondition(space.Status.Conditions, terminatingSpace) {
+			// space is in terminating let's wait until is gone and recreate it ...
+			return false, nil
+		}
```

Comment:
let's rather rely on the deletion timestamp than the condition.
and you can right away wait until it's gone
```suggestion
		if util.IsBeingDeleted(space) {
			// space is in terminating let's wait until is gone and recreate it ...
			return false, a.WaitUntilSpaceAndSpaceBindingsDeleted(t, space.Name)
		}
```

Reply:
Nice ! applied in https://github.com/codeready-toolchain/toolchain-e2e/pull/689/commits/6810c96b1fd58e73f4637c474d4a36135a1419c7

Thanks!